src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/TelemetryUtility.cs

```diff
@@ -21,5 +24,148 @@ public static string CreateFileAndForgetEventName(string typeName, string member
 
             return $"{VSTelemetrySession.VSEventNamePrefix}fileandforget/{typeName}/{memberName}";
         }
+
+        /// <summary>
+        /// True if the source is http and ends with index.json
+        /// </summary>
+        public static bool IsHttpV3(PackageSource source)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            return source.IsHttp &&
+                (source.Source.EndsWith("index.json", StringComparison.OrdinalIgnoreCase)
+                || source.ProtocolVersion == 3);
+        }
+
+        /// <summary>
+        /// True if the source is HTTP and has a *.nuget.org or nuget.org host.
+        /// </summary>
+        public static bool IsNuGetOrg(PackageSource source)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            if (!source.IsHttp)
+            {
+                return false;
+            }
+
+            var uri = source.TrySourceAsUri;
+            if (uri == null)
+            {
+                return false;
+            }
+
+            if (StringComparer.OrdinalIgnoreCase.Equals(uri.Host, "nuget.org")
+                || uri.Host.EndsWith(".nuget.org", StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// True if the source is an Azure Artifacts (DevOps) feed
+        /// </summary>
+        public static bool IsAzureArtifacts(PackageSource source)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            if (!source.IsHttp)
+            {
+                return false;
+            }
+
+            var uri = source.TrySourceAsUri;
+            if (uri == null)
+            {
+                return false;
+            }
```

Comment:
I think we can move these checks to a different method as they appear same in `IsAzureArtifacts`, `IsNuGetOrg`, `IsGitHub' methods.